source/ezrassor_topic_switch/parsing.py

```diff
@@ -0,0 +1,87 @@
+"""Parse arguments from the command line.
+
+references:
+  docs.python.org/3/library/argparse.html
+"""
+import argparse
+import importlib
+
+
+def load_message_type(raw_args, program_name):
+    """Load a message type from its corresponding Python module."""
+
+    class ArgumentParser(argparse.ArgumentParser):
+        """Parse revelant arguments using argparse."""
+
+        def __init__(self, *args, **keyword_args):
+            """Initialize the parser with the --message-type argument."""
+            super().__init__(
+                usage="\n $ %(prog)s [OPTIONS]",
+                *args,
+                **keyword_args,
+            )
+
+            self.add_argument(
+                "--message-type",
+                help="topic message type, like 'std_msgs.msg.String'",
+                metavar="TYPE",
+                required=True,
+            )
+
+        def error(self, message):
+            """Throw an error if something goes wrong.
+
+            By default, argparse ArgumentParsers print a help blurb directly to
+            the console if something goes wrong. This strategy doesn't obey the
+            functional paradigm (no opportunity to handle the error, breaks the
+            call chain, creates a side effect, etc). We throw an error instead
+            that we handle upstream.
+            """
+            raise ParseError(message, self.format_help())
+
+    parser = ArgumentParser(prog=program_name)
+    parsed_args = vars(parser.parse_known_args(raw_args)[0])
+
+    try:
+        module_name, message_name = parsed_args["message_type"].rsplit(".", 1)
+    except ValueError:
+        raise MessageLoadError(
+            "message type must include periods, like 'std_msgs.msg.String'",
+        )
+
+    try:
+        module = importlib.import_module(module_name)
+    except ImportError:
+        raise MessageLoadError(f"no module named '{module_name}'")
+
+    try:
+        return vars(module)[message_name]
+    except KeyError:
+        raise MessageLoadError(
+            f"no message named '{message_name}' in module '{module_name}'"
+        )
+
+
+class ParseError(Exception):
+    """Encapsulate parsing-related errors."""
+
+    def __init__(self, message, usage):
+        """Initialize this error with a message and a usage blurb."""
+        self.message = message
+        self.usage = usage
+
+    def __str__(self):
+        """Create a human-readable representation of this error."""
+        return f"error:\n  {self.message}\n\n{self.usage}"
```

Comment:
Is ParseError a type of Exception? Shouldn't it be called ParseException then?

Reply:
As discussed offline, Python standard seems to be that exceptions are suffixed with "Error"